### DISCUSSION TITLE
Firebase admin not initializing

### DISCUSSION THREAD

**Discussion by n-zu** [gladly-team/next-firebase-auth#480] [Q&A]
I'm trying to use firebase-admin before `setAuthCookies` but it appears as not initialized.

`login.js`
```
import { setAuthCookies } from "next-firebase-auth";
import initAuth from "../../server/initAuth";
import { getAuth } from "firebase-admin/auth";

initAuth();

const isAdmin = (user) => {
  return user?.role === "admin";
};

const handler = async (req, res) => {
  try {
    const token = req?.headers?.authorization;
    if (!token) return res.status(403).json({ error: "Unauthorized" });
    const user = await getAuth().verifyIdToken(token);

    if (!isAdmin(user)) return res.status(403).json({ error: "Unauthorized" });

    await setAuthCookies(req, res);
  } catch (e) {
    return res.status(500).json({ error: "Unexpected error.", e });
  }
  return res.status(200).json({ success: true });
};

export default handler;
```
`error`
```
{
  "code": "app/no-app",
  "message": "The default Firebase app does not exist. Make sure you call initializeApp() before using any of the Firebase services."
}
```

if in dev mode i comment a out the code, login with an user and put the code back then it works as expected,
my assumption is that the admin app is not initialized until the first call to `setAuthCookies`

is there a way to initialize it beforehand ?

**Comment by kmjennison** [comment]
Please try out the [example app](https://github.com/gladly-team/next-firebase-auth/tree/main/example). It has an API endpoint similar to what you're looking to do: https://github.com/gladly-team/next-firebase-auth/blob/main/example/pages/api/example.js

If that doesn't help identify the problem, please provide a reproduction of the issue with an example app. Thanks!

**Reply by n-zu** [reply]
Thanks !
I'll check it out
